### ISSUE TITLE
AttributeError: 'AppserviceServer' object has no attribute 'collect'

### ISSUE THREAD

**Issue by turt2live** [matrix-org/synapse#3357]
```
appservice - 2018-06-06 17:52:46,385 - twisted - 131 - CRITICAL - -
Traceback (most recent call last):
  File "/home/matrix/.synapse/local/lib/python2.7/site-packages/twisted/web/server.py", line 197, in process
    self.render(resrc)
  File "/home/matrix/.synapse/local/lib/python2.7/site-packages/synapse/http/site.py", line 83, in render
    return Request.render(self, resrc)
  File "/home/matrix/.synapse/local/lib/python2.7/site-packages/twisted/web/server.py", line 257, in render
    body = resrc.render(self)
  File "/home/matrix/.synapse/local/lib/python2.7/site-packages/twisted/web/resource.py", line 250, in render
    return m(request)
  File "/home/matrix/.synapse/local/lib/python2.7/site-packages/prometheus_client/twisted/_exposition.py", line 18, in render_GET
    return generate_latest(self.registry)
  File "/home/matrix/.synapse/local/lib/python2.7/site-packages/prometheus_client/exposition.py", line 69, in generate_latest
    for metric in registry.collect():
AttributeError: 'AppserviceServer' object has no attribute 'collect'
```

Unknown cause. Synapse 0.31.0 on t2bot.io

**Comment by silkeh** [comment]
Can confirm this is an issue for the appservice worker (`synapse.app.appservice`).

Seems to me that the cause is the lack of collect method for the Prometheus scraper. Disabling metrics for this endpoint fixes the startup of the worker, but results in a lack of monitoring.

**Comment by richvdh** [comment]
weird, it's working here :/

**Comment by hawkowl** [comment]
This should now be fixed by #3390. Thanks @turt2live !
